diff -Naur v0.4/src/display/device/PangolinNSGLView.mm patch/src/display/device/PangolinNSGLView.mm
--- v0.4/src/display/device/PangolinNSGLView.mm	2016-08-08 21:56:24.000000000 -0700
+++ patch/src/display/device/PangolinNSGLView.mm	2016-09-10 21:50:12.000000000 -0700
@@ -77,6 +77,11 @@
 #endif
         backing_scale = 1.0;

+    // This method may be called from a separate (Core Animation) thread, so the thread-local pangolin::context may be
+    // uninitialized.
+    if (pangolin::context == 0) {
+        pangolin::context = context;
+    }
     pangolin::process::Resize(self.bounds.size.width * backing_scale, self.bounds.size.height * backing_scale);

     [[self openGLContext] update];
diff -Naur v0.4/src/CMakeLists.txt patch/src/CMakeLists.txt
--- v0.4/src/CMakeLists.txt	2016-08-08 21:56:24.000000000 -0700
+++ patch/src/CMakeLists.txt	2016-10-05 19:11:00.000000000 -0700
@@ -374,35 +374,6 @@
   message(STATUS "Pleora Found and Enabled" )
 endif()

-if(NOT BUILD_EXTERN_LIBPNG)
-  find_package(PNG QUIET)
-endif()
-if(PNG_FOUND)
-  # (ZLIB is also found by FindPNG.cmake as its dependency)
-  set(HAVE_PNG 1)
-  list(APPEND INTERNAL_INC ${PNG_INCLUDE_DIR} )
-  list(APPEND LINK_LIBS ${PNG_LIBRARY} ${ZLIB_LIBRARY} )
-  message(STATUS "libpng Found and Enabled")
-endif()
-
-if(NOT BUILD_EXTERN_LIBJPEG)
-  find_package(JPEG QUIET)
-endif()
-if(JPEG_FOUND)
-  set(HAVE_JPEG 1)
-  list(APPEND INTERNAL_INC ${JPEG_INCLUDE_DIR} )
-  list(APPEND LINK_LIBS ${JPEG_LIBRARY} )
-  message(STATUS "libjpeg Found and Enabled")
-endif()
-
-find_package(TIFF QUIET)
-if(TIFF_FOUND)
-  set(HAVE_TIFF 1)
-  list(APPEND INTERNAL_INC ${TIFF_INCLUDE_DIR} )
-  list(APPEND LINK_LIBS ${TIFF_LIBRARY} )
-  message(STATUS "libtiff Found and Enabled")
-endif()
-
 find_package(OpenEXR QUIET)
 if(OpenEXR_FOUND)
   set(HAVE_OPENEXR 1)
diff -Naur v0.4/CMakeModules/FindGLEW.cmake patch/CMakeModules/FindGLEW.cmake
--- v0.4/CMakeModules/FindGLEW.cmake	2016-08-08 21:56:24.000000000 -0700
+++ patch/CMakeModules/FindGLEW.cmake	2016-10-05 19:14:59.000000000 -0700
@@ -27,7 +27,7 @@
         /opt/local/include
         DOC "The directory where GL/glew.h resides")
     FIND_LIBRARY( GLEW_LIBRARY
-        NAMES GLEW glew
+        NAMES libglew.a GLEW glew
         PATHS
         /usr/lib64
         /usr/lib
